debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/TransactionCommitConsumer.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.oracle.logminer.processor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.connector.oracle.BlobChunkList;
+import io.debezium.connector.oracle.OracleConnectorConfig;
+import io.debezium.connector.oracle.OracleDatabaseSchema;
+import io.debezium.connector.oracle.logminer.LogMinerHelper;
+import io.debezium.connector.oracle.logminer.events.DmlEvent;
+import io.debezium.connector.oracle.logminer.events.EventType;
+import io.debezium.connector.oracle.logminer.events.LobWriteEvent;
+import io.debezium.connector.oracle.logminer.events.LogMinerEvent;
+import io.debezium.connector.oracle.logminer.events.SelectLobLocatorEvent;
+import io.debezium.function.BlockingConsumer;
+import io.debezium.relational.Table;
+
+/**
+ * A consumer of transaction events at commit time that is capable of inspecting the event stream,
+ * merging events that should be merged when LOB support is enabled, and then delegating the final
+ * stream of events to a delegate consumer.
+ *
+ * @author Chris Cranford
+ */
+public class TransactionCommitConsumer implements AutoCloseable, BlockingConsumer<LogMinerEvent> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TransactionCommitConsumer.class);
+
+    private final BlockingConsumer<LogMinerEvent> delegate;
+    private final OracleConnectorConfig connectorConfig;
+    private final OracleDatabaseSchema schema;
+    private final List<String> lobWriteData;
```

Comment:
What is this about?

Reply:
So this is a reimagined implementation of the `TransactionReconciliation` class that was removed.  When the connector is configured with LOB support, we get multiple events for the same logical SQL statement, i.e.:

```
INSERT INTO my_table (id,lob_field) values (1, 'some lob data');
```
would be provided by LogMiner as:
```
INSERT INTO my_table (id,lob_field) values (1, EMPTY_CLOB());
UPDATE my_table SET lob_field = 'some lob data' WHERE id = 1;
````

We didn't want to emit this as two separate DML events but rather a single INSERT since that is what the user did.  So this class acts as a buffered consumer of events at commit time, tracking the previous event and delaying its emission as a change event based on certain criteria.  

So in the above example, the `INSERT` would get buffered, the `UPDATE` would then be consumed and would get merged into the `INSERT` and then when the commit process finishes since there was only 1 logical event, the closing of the consumer triggers the emission of the last buffered event, the `INSERT` which would for all intents and purposes resemble the logical SQL statement the user issued:

```
"after": {
  "id": 1,
  "lob_field": "some lob data"
}
```

The reason why the reconciliation class was reimagined had all to do with cache interactions.  In the prior impl, it did a lot of shuffling in the `List<LogMinerEvent>` collection, removing events after they had been merged into the prior event, etc.  I wanted to avoid this because previously we were iterating the events twice in the collection at commit time and there really was no need for that.  Additionally, this allows us to reduce the ISPN cache interactions and keep a common approach across both impls.